### DISCUSSION TITLE
writing a calculation for an even split

### DISCUSSION THREAD

**Discussion by d-opri** [neuefische/questions#475] [Web] [answered]
moin moin,

i am writing a function that should calculate an even split between my participants, one of them is checked as having paid, so they would receive money and the rest should owe their share. for that the function should receive: the participants (an array), the total (a number) and a boolean 'paidBy'. the idea is to go through every participant, and if paidBy is true, they will have their share subtracted from the total, and for the rest they will have the even split of the total, shown as a negative.
````
function calculateSplitShare(participants, amount, paidBy) {
  const splitShare = participants.map((participant) => {
    paidBy
      ? {
          id: participant.id, 
          name: participant.label,
          share: amount - (amount / participant.length),
        }
      : {
          id: participant.id,
          name: participant.label,
          share: -amount / participant.length,
        };
  });
  return splitShare;
}
````
i want to write a test but i am unsure how to go about it, 
```
describe("Calculate Split Function", () => {
  test("it should divide amount equally between the number of 
participants and return positive share if 'paidBy' and negative if not", () => {


// i am unsure how to expect my result, but since i would also 
need different results (the ones in debt and the one owed),
or do i need to write a second test?

    expect(calculateSplitShare(participants, amount)).toBe();
  });
});
```

i really dont know how to connect the calculation with the inputs from my form, and also if i even got it right...any help would be much appreciated 🎱

**Comment by F-Kirchhoff** [comment, marked as answer]
First of all, the "paidBy" boolean won't work if you map internally. Since if the boolean is "true", the first case will be returned for all participants, and if its false the last one. The 'paidBy' parameter needs to distinguish between the participants. For Example, it can be the user ID of the paying participant. Then you can check in your map if user.id === paidById (a rename might be useful here) and by that find the user who should be handled differently.
For the testing. You should build a very simple example where you already know the outcome, for example two participants, one paid 20 dollar. Then you know that the expected outcome should be something like this:
```
[{
  user.id: 1,
  share: 10,
},{
user.id: 2,
share: -10,
}]
```
Then you need to hard code this outcome and paste it into the `.toBe` of the testing. Be sure you format it correctly and that every character is as it should be. You need to create the array, amount and paidById, which you pass to the function, as well. 

I hope that was helpful.
